### PR TITLE
Replace references to Replication Controllers with Deployments in doc…

### PR DESCRIPTION
**Pull request by pwittrock**
…s/user-guide/walkthrough.

Also:
* Cleanup pasted yaml and instead embed file
* Clarify Service test sections by including busybox exec commands as part of the preformatted section

### CODE REVIEW

**Comment by pwittrock** [comment]
I am wondering if this example is not going to work since the docs still reference the k/k repo, but the yaml is only in the k/k.g.i repo.

File: docs/user-guide/walkthrough/index.md [hunk 1]
```diff
@@ -64,10 +53,12 @@ $ kubectl get pods
 
 On most providers, the pod IPs are not externally accessible. The easiest way to test that the pod is working is to create a busybox pod and exec commands on it remotely. See the [command execution documentation](/docs/user-guide/getting-into-containers/) for details.
 
-Provided the pod IP is accessible, you should be able to access its http endpoint with curl on port 80:
+Provided the pod IP is accessible, you should be able to access its http endpoint with wget on port 80:
 
 ```shell
-$ curl http://$(kubectl get pod nginx -o go-template={{.status.podIP}})
+$ kubectl run busybox --image=busybox --restart=Never --tty -i  --env "POD_IP=$(kubectl get pod nginx -o go-template={{.status.podIP}})"
```
Comment (janetkuo) [inline docs/user-guide/walkthrough/index.md]: If you want to generate pods here, use `--generator=run-pod/v1` and `--restart=Never` is optional.

Comment (janetkuo) [inline docs/user-guide/walkthrough/index.md]: I just found that `{{.status.podIP}}` was escaped on the website so the users won't be able to see and copy the right command. @johndmulhausen

Comment (pwittrock) [inline docs/user-guide/walkthrough/index.md]: Do you think using the pod generator is better?  It has the advantage that we don't need to delete the jobl

Comment (pwittrock) [inline docs/user-guide/walkthrough/index.md]: `{{.status.podIP}}` doesn't seem to be escaped when I hit view.  If I copy and paste the commands, they produce the desired result.  (getting the nginx page)

Comment (janetkuo) [inline docs/user-guide/walkthrough/index.md]: It doesn't render well on the website http://kubernetes.io/docs/user-guide/walkthrough/#pod-management
screenshot from 2016-03-23 13 33 49

Comment (janetkuo) [inline docs/user-guide/walkthrough/index.md]: > Do you think using the pod generator is better? It has the advantage that we don't need to delete the job.

Yes and plus we don't need to introduce/explain what a "job" means.

**Comment by pwittrock** [comment]
@johndmulhausen I am adding new yaml config files for the deployment, but this guide has the user get the files from a release or the k/k example.  I am probably going to need to convert this example to use an entirely different convention based off the files in the k/k.g.i repo.  The click-to-copy addition to the widget is nice.  I'd like to explore if curling the file as part of the command would further simplify things for the user.  Does a widget exist that I could use to directly reference the absolute url to a raw yaml file in the current repo/branch or is this something that would be simple to create?

**Comment by pwittrock** [comment]
For now I am just telling the user to copy paste these files.  Definitely breaks up the flow since the other files are assumed to be checked out, but at least it is funcitonal.

**Comment by pwittrock** [comment]
Found a couple issues, fixing now

**Comment by pwittrock** [comment]
Fixed up and published to http://pwittrock.github.io/docs/user-guide/walkthrough/k8s201/.  PTAL

File: docs/user-guide/walkthrough/k8s201.md [hunk 2]
```diff
@@ -9,114 +9,143 @@ scaling.
...
-
-{% include code.html language="yaml" file="replication-controller.yaml" ghlink="/docs/user-guide/walkthrough/replication-controller.yaml" %}
+List all deployments:
 
+```shell
+kubectl get deployment
+```
 
-#### Replication Controller Management
+List the pods created by the deployment:
```
Comment (janetkuo) [inline docs/user-guide/walkthrough/k8s201.md]: The shell block isn't rendered well on the website. Perhaps adding a new line can solve this?
http://pwittrock.github.io/docs/user-guide/walkthrough/k8s201/#deployment-management
List the pods created by the deployment: `shell kubectl get pods -l app=nginx`

File: docs/user-guide/walkthrough/k8s201.md [hunk 3]
```diff
@@ -9,114 +9,143 @@ scaling.
...
+```shell
+kubectl get pods -l app=nginx
+```
 
-Create an nginx replication controller ([replication-controller.yaml](/docs/user-guide/walkthrough/replication-controller.yaml)):
+Upgrade the nginx container from 1.7 to 1.8 by changing the deployment and calling `apply`.  The following config
```
Comment (janetkuo) [inline docs/user-guide/walkthrough/k8s201.md]: from 1.7.9 to 1.8 (here and the comment in yaml file)

**Comment by janetkuo** [comment]
Had a few comments

File: docs/user-guide/walkthrough/k8s201.md (hunk 2 above)
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: Done

File: docs/user-guide/walkthrough/k8s201.md (hunk 3 above)
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: Done

**Comment by pwittrock** [comment]
Thnx PTAL

**Comment by pwittrock** [comment]
squashed commits

File: docs/user-guide/walkthrough/index.md [hunk 4]
```diff
@@ -64,11 +53,12 @@ $ kubectl get pods
 
 On most providers, the pod IPs are not externally accessible. The easiest way to test that the pod is working is to create a busybox pod and exec commands on it remotely. See the [command execution documentation](/docs/user-guide/getting-into-containers/) for details.
 
-Provided the pod IP is accessible, you should be able to access its http endpoint with curl on port 80:
+Provided the pod IP is accessible, you should be able to access its http endpoint with wget on port 80:
```
Comment (pwittrock) [inline docs/user-guide/walkthrough/index.md]: The wget is actually to be run in the busybox container (which has wget, but doesn't have curl).  Is there a syntax for representing running the command within the container?  e.g. `pod@busyboxy$ wget -q0- http://$POD_IP`?

Comment (johndmulhausen) [inline docs/user-guide/walkthrough/index.md]: Oh, gotcha! Yeah, I think that syntax looks fine. The shell usually gives you somesuch preamble.

File: docs/user-guide/walkthrough/k8s201.md [hunk 5]
```diff
@@ -9,114 +9,144 @@ scaling.
...
+{% include code.html language="yaml" file="deployment.yaml" ghlink="/docs/user-guide/walkthrough/deployment.yaml" %}
+
 
+#### Deployment Management
 
-## Replication Controllers
+Create an nginx deployment ([deployment.yaml](/docs/user-guide/walkthrough/deployment.yaml)):
 
-OK, now you know how to make awesome, multi-container, labeled pods and you want to use them to build an application, you might be tempted to just start building a whole bunch of individual pods, but if you do that, a whole host of operational concerns pop up.  For example: how will you scale the number of pods up or down and how will you ensure that all pods are homogeneous?
+Copy this to a loca file named `deployment.yaml`
```
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: No longer applicable.

File: docs/user-guide/walkthrough/k8s201.md [hunk 6]
```diff
@@ -9,114 +9,144 @@ scaling.
...
+
+Here is a deployment that instantiates two nginx pods:
+
+{% include code.html language="yaml" file="deployment.yaml" ghlink="/docs/user-guide/walkthrough/deployment.yaml" %}
+
 
+#### Deployment Management
 
-## Replication Controllers
+Create an nginx deployment ([deployment.yaml](/docs/user-guide/walkthrough/deployment.yaml)):
```
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: Updated to tell the user to download from the widget above.

File: docs/user-guide/walkthrough/k8s201.md [hunk 7]
```diff
@@ -9,114 +9,144 @@ scaling.
...
+kubectl get pods -l app=nginx
+```
+
+Delete the deployment by name:
+
+```shell
+kubectl delete deployment nginx-deployment
+```
+
+For more information, such as how to rollback deployment changes to a previous version, see [_Deployments_](/docs/user-guide/deployment.md).
```
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: done

File: docs/user-guide/walkthrough/k8s201.md [hunk 8]
```diff
@@ -9,114 +9,144 @@ scaling.
...
 For more information, see [Labels](/docs/user-guide/labels).
-They are a core concept used by two additional Kubernetes building blocks: Replication Controllers and Services.
+They are a core concept used by two additional Kubernetes building blocks: Deployments and Services.
+
+
+## Deployments
+
+OK, now you know how to make awesome, multi-container, labeled pods and you want to use them to build an application, you might be tempted to just start building a whole bunch of individual pods, but if you do that, a whole host of operational concerns pop up.  For example: how will you scale the number of pods up or down?  how will you rollout a new release?
+
+The answer to those questions and more is to use a [_Deployment_](/docs/user-guide/deployment.md) to manage maintaining and updating your running _Pods_.
```
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: Done

**Comment by pwittrock** [comment]
PTAL.  In the published version, the click-to-copy button and download link do not seem to be working.  Is the new widget the default?

**Comment by janetkuo** [comment]
It failed on master as well. @johndmulhausen is fixing it.

**Comment by janetkuo** [comment]
It's fixed now.

**Comment by pwittrock** [comment]
I'll republish this now that is my only open docs pr

File: docs/user-guide/walkthrough/k8s201.md [hunk 9]
```diff
@@ -1,122 +1,153 @@
...
+kubectl get pods -l app=nginx
+```
+
+For more information, see [Labels](/docs/user-guide/labels/).
+They are a core concept used by two additional Kubernetes building blocks: Deployments and Services.
+
+
+## Deployments
+
+OK, now you know how to make awesome, multi-container, labeled pods and you want to use them to build an application, you might be tempted to just start building a whole bunch of individual pods, but if you do that, a whole host of operational concerns pop up.  For example: how will you scale the number of pods up or down?  how will you rollout a new release?
```
Comment (johndmulhausen) [inline docs/user-guide/walkthrough/k8s201.md]: s/how will you roll out/How will you roll out
s/OK, now you know/Now that you know

File: docs/user-guide/walkthrough/k8s201.md [hunk 10]
```diff
@@ -1,122 +1,153 @@
...
+
+The answer to those questions and more is to use a [_Deployment_](/docs/user-guide/deployment/) to manage maintaining and updating your running _Pods_.
+
+A deployment object defines a pod creation template (a "cookie-cutter" if you will) and desired replica count.  The deployment uses a label selector to identify the pods it manages, and will create or delete pods as needed to meet the replica count.  Deployments are also used to manage safely rolling out changes to your running pods.
```
Comment (johndmulhausen) [inline docs/user-guide/walkthrough/k8s201.md]: Should Deployments and Pods be capitalized?

File: docs/user-guide/walkthrough/k8s201.md [hunk 11]
```diff
@@ -1,122 +1,153 @@
...
+Once you have a replicated set of pods, you need an abstraction that enables connectivity between the layers of your application.  For example, if you have a deployment managing your backend jobs, you don't want to have to reconfigure your front-ends whenever you re-scale your backends.  Likewise, if the pods in your backends are scheduled (or rescheduled) onto different machines, you can't be required to re-configure your front-ends.  In Kubernetes, the service abstraction achieves these goals.  A service provides a way to refer to a set of pods (selected by labels) with a single static IP address. It may also provide load balancing, if supported by the provider.
```
Comment (johndmulhausen) [inline docs/user-guide/walkthrough/k8s201.md]: Should Deployments, Jobs, and Pods be capitalized?

**Comment by johndmulhausen** [comment]
@pwittrock A couple nits, but the main thing is we have merge conflicts now. :(

File: docs/user-guide/walkthrough/k8s201.md (hunk 9 above)
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: Done

File: docs/user-guide/walkthrough/k8s201.md (hunk 10 above)
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: Done

File: docs/user-guide/walkthrough/k8s201.md (hunk 11 above)
Comment (pwittrock) [inline docs/user-guide/walkthrough/k8s201.md]: Done

**Comment by pwittrock** [comment]
PTAL

**Comment by pwittrock** [comment]
@johndmulhausen Let me know when you are free to review this and then I will resolve the merge conflicts.

**Comment by bgrant0607** [comment]
@pwittrock @janetkuo Let's not wait for @johndmulhausen's review

**Comment by pwittrock** [comment]
SGTM  @janetkuo Do you want to review again?  Can I self merge?

**Comment by johndmulhausen** [comment]
Looks good to me. :) Thanks!

**Pull request merged**